### PR TITLE
Addition of saturated flag in SiStrip Approximate Clusters

### PR DESCRIPTION
**Pull request by abaty**
This PR slightly extends the SiStripApproximateCluster data format to include a boolean flag which marks clusters which are saturated.  This change was motivated by the discussion at [1].  The definition of 'saturated' we use here is the same as what is used in the SubClusterShapeFilter, i.e. a cluster having 3 consecutive strips of ADC>=254.  Note that this flag is not currently used in the reconstruction sequence; this PR is simply to ensure this information is not dropped in the new approx. cluster data format.  This is expected to increase the approximate cluster collection size by ~1% and barely affect the timing performance [2].

[1] https://github.com/cms-sw/cmssw/pull/38423#issuecomment-1168993837
[2] https://twiki.cern.ch/twiki/pub/Main/HIDetectorReadout2020/SaturatedStrips.pdf

#### PR validation:
This PR was tested by running workflow 140.58 step 2, and checking the approximated cluster collection in the output.  

@mandrenguyen @icali

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflow = 140.58

**Comment by mmusich** [comment]
please test

File: DataFormats/SiStripCluster/src/classes_def.xml [hunk 1]
```diff
@@ -24,8 +24,8 @@
  <class name="edm::Wrapper<edmNew::DetSetVector<edm::Ref<edmNew::DetSetVector<SiStripCluster>,SiStripCluster,edmNew::DetSetVector<SiStripCluster>::FindForDetSetVector> > >" />
 
 
- <class name="SiStripApproximateCluster" ClassVersion="3">
-  <version ClassVersion="3" checksum="2041370183"/>
+ <class name="SiStripApproximateCluster" ClassVersion="9">
+  <version ClassVersion="9" checksum="2854791577"/>
```
Comment (jpata) [inline DataFormats/SiStripCluster/src/classes_def.xml:28]: why did it jump from 3 -> 9?

**Comment by jpata** [comment]
The size increase overall seems negligible, so once the class version is sorted out, it's good to go from my side.

```
140.58, 500 events, file size in bytes
step2.root: 2553402430 -> 2559880253
step3.root: 1159401041 -> 1159380736
```

File: DataFormats/SiStripCluster/src/classes_def.xml (hunk 1 above)
Comment (abaty) [reply DataFormats/SiStripCluster/src/classes_def.xml]: I have updated this file to be version '4' now.

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
+reconstruction
- changes the SiStrip approximate cluster dataformat for HI
- file size increase is minimal
- no reco changes

**Comment by qliphy** [comment]
+1

File: DataFormats/SiStripCluster/src/classes_def.xml [hunk 2]
```diff
@@ -24,8 +24,8 @@
  <class name="edm::Wrapper<edmNew::DetSetVector<edm::Ref<edmNew::DetSetVector<SiStripCluster>,SiStripCluster,edmNew::DetSetVector<SiStripCluster>::FindForDetSetVector> > >" />
 
 
- <class name="SiStripApproximateCluster" ClassVersion="3">
-  <version ClassVersion="3" checksum="2041370183"/>
+ <class name="SiStripApproximateCluster" ClassVersion="4">
+  <version ClassVersion="4" checksum="2854791577"/>
```
Comment (makortel) [inline DataFormats/SiStripCluster/src/classes_def.xml:28]: This should have kept the class version for version 3.

Comment (jpata) [reply DataFormats/SiStripCluster/src/classes_def.xml:28]: Thanks Matti. Is there a way to detect this at compile time?

Comment (makortel) [reply DataFormats/SiStripCluster/src/classes_def.xml:28]: Nothing straightforward comes to my mind, but maybe we could figure out something. A specific check would certainly help to catch such cases. Could you open an issue about it?

Comment (jpata) [reply DataFormats/SiStripCluster/src/classes_def.xml:28]: done! https://github.com/cms-sw/cmssw/issues/39008

**Pull request merged**